docs/src/main/asciidoc/deploying-to-openshift-howto.adoc

```diff
@@ -0,0 +1,113 @@
+:extensions: io.quarkus:quarkus-openshift
+
+As an application developer, you build and deploy your {project-name} applications to {openshift} by using a single command.
+
+With this one command, you build your application locally, trigger a container image build, then apply the generated {openshift} resources automatically.
+The generated resources use a Kubernetes `Deployment`, but still make use of OpenShift-specific resources, such as `Route`, `BuildConfig`, and so on.
+
+[IMPORTANT]
+====
+As of {openshift} 4.14, the `DeploymentConfig` object is deprecated.
+`Deployment` is now the default and preferred deployment kind for the the `quarkus-openshift` extension.
+
+Because of this change, consider the following:
+
+* If you redeploy applications that you previously deployed by using `DeploymentConfig`, by default, those applications use `Deployment` but do not remove the previous `DeploymentConfig`.
+This leads to a deployment of both new and old applications, so, you must remove the old `DeploymentConfig` manually.
+If you want to continue to use `DeploymentConfig`, it is still possible to do so by explicitly setting `quarkus.openshift.deployment-kind` to `DeploymentConfig`.
+* `Deployment` is a Kubernetes resource and not OpenShift specific, so it cannot leverage `ImageStream` resources, as is the case with `DeploymentConfig`.
+Therefore, the image references must include the container image registry that hosts the image.
+
+For more information about the deprecation, how to set up and use automatic rollbacks, triggers, lifecycle hooks, and custom strategies, see the Red Hat Knowledgebase article https://access.redhat.com/articles/7041372[DeploymentConfig API is being deprecated in Red Hat OpenShift Container Platform 4.14].
+====
+
+== Prerequisites
+
+* You have OpenJDK {jdk-version-other} or later installed.
+* You have set the `JAVA_HOME` environment variable to the location of the Java SDK.
+* You have Apache Maven {maven-version} installed.
+* You have access to an {openshift} cluster and the latest compatible version of the `oc` CLI tool installed.
+* You are working in the correct OpenShift project namespace.
+
+[NOTE]
+====
+To run this deployment, it is not necessary to have the `quarkus-openshift` extension included in your {project-name} project.
+====
+
+== Build and deploy a {project-name} application on OpenShift
+
+. Trigger a build and deployment by running the following command:
+
+:build-additional-parameters: quarkus deploy openshift
+include::{includes}/devtools/build.adoc[]
+:!build-additional-parameters:
+
+[TIP]
+====
+If you want to test your application immediately, set the `quarkus.openshift.route.expose` configuration property to `true` to expose the service automatically.
+For example, add `-Dquarkus.openshift.route.expose=true` to the above command.
+For more information, see xref:deploying-to-openshift.adoc#exposing_routes[Exposing routes].
+====
+
+[[re-deploy-with-service-binding]]
```

Comment:
I don't think that we should have any reference to service binding in the `deploy in a single step` doc.

Reply:
Removed paragraph. Note, this service binding reference was in the existing upstream guide. Should it be added elsewhere?